src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.Caching.cs

```diff
@@ -561,6 +562,7 @@ public int GetHashCode(JsonSerializerOptions options)
                 AddHashCode(ref hc, options._unmappedMemberHandling);
                 AddHashCode(ref hc, options._defaultBufferSize);
                 AddHashCode(ref hc, options._maxDepth);
+                AddHashCode(ref hc, options._newLine ?? Environment.NewLine); // Null is equivalent to the default
```

Comment:
You could try calling the property directly to avoid replicating the default value here.